# Homework 4, Assignment 3

In [9]:
import os
import sys
import numpy as np
import pylab as pl


import json
import os

%pylab inline

Populating the interactive namespace from numpy and matplotlib


# Null Hypothesis:

## Idea:
Those who convicted felonies are less likely to revert back to criminal behavior if they participate in the employment program.

## Null Hypothesis
The percentage of former convicted felons who would recitivate 1-3 years after release from prison is the same or higher for employment program participants than as for the control group, significance level p=0.05

## Alternative Hypothesis
The percentage of former convicted felons who would recitivate 1-3 years after release from prison is lower for employment program participants than as for the control group, significance level p=0.05

H0 = P_1 - P_0 <= 0

Ha = P_0 - P_1 > 0


#### Notations:
P_0 = Program group percentage, 

P_1 = Control group percentage, 

n_0 = Program group population, 

n_1 = Cotrol group poplulation

sp = sample proportion

In [54]:
a = 0.05 #where a is alpha

P_0 = 10 * 0.01 #Program group
P_1 = 11.7 * 0.01 #Control group

if P_1 - P_0 <= 0:
    print ("Do not reject null")
else:
    print ("we must assess the statistical significance")
    
n_0 = 568
n_1 = 409


Nt_0 = P_0 * n_0
Nt_1 = P_1 * n_1

we must assess the statistical significance


## Z-Test

In [31]:
#sp = Standard Proportion
sp = (P_0 * n_0 + P_1 * n_1) / (n_1 + n_0)
print (sp)

0.107116683726


In [35]:
#se = Standard Error
p = lambda P_0, P_1, n_0, n_1: (P_0 * n_0 + P_1 * n_1) / (n_0 + n_1)

se = lambda p, n_0, n_1: np.sqrt(p * (1 - p) * (1.0 / n_0 + 1.0 / n_1)) 

## z-score

Calculates how many standard deviations away from the the population parameters my statistic is

In [36]:
zscore = lambda P_0, P_1, s : (P_0 - P_1) / s 

In [40]:
z_2y = zscore(P_0, P_1, se(p(P_0, P_1, n_0, n_1), n_0, n_1))
print (z_2y)

-0.84764020522


In [47]:
p_2y = 1 - 0.7995


def report_result(p,a):
    print ('is the p value ' + 
           '{0:.2f} smaller than the critical value {1:.2f}?'.format(p,a))
    if p < a:
        print ("YES!")
    else: 
        print ("NO!")
    
    print ('the Null hypothesis is {}'.format(\
                            'rejected' if p < a  else 'not rejected') )

    
report_result(p_2y, a) #where a is alpha

is the p value 0.20 smaller than the critical value 0.05?
NO!
the Null hypothesis is not rejected


### Because the p value of 0.20 is NOT smaller than the significance value (alpha), the Null Hypothesis is NOT rejected

# CHI SQ

|Recitivated |     YES         | NO     |                   
|---------------------------|:---------------------:|------------------|---------------------------|
| program sample               | $0.10*568$           | $0.90*568$      | 568                       |
| control sample            | $0.117*409$           | $.883*409$     | 409                       |
|                           |                       |                  |                           |
| total                     | 104.653               |  872.347        | 977      

In [48]:
def evalChisq(values):
    values = np.array(values)
    E = np.empty_like(values)
    for j in range(len(values[0])):
        for i in range(2):
            
            E[i][j] = ((values[i,:].sum() * values[:,j].sum()) / 
                        (values).sum())
    return ((values - E)**2 / E).sum()

In [49]:
sample_values = np.array([[0.10 * 568,  0.900 * 568], [0.117 * 409, 0.883 * 409]])
print (evalChisq(sample_values))

0.718493917505


### Degrees of Freedom = Number of Observations - number of independent variables

### 1 = 2 - 1 (DOF = 1)

### When referring to the CHI SQ table, a CHI SQ statistic of ~.7185 produces a minimum CHI SQ significanct value of 3.84. Thus we DO NOT reject the Null Hypothesis. 